src/lua/BabblerBomb/Bombler.lua

```diff
@@ -0,0 +1,284 @@
+
+class 'Bombler' (Babbler)
+
+Bombler.kMapName = "bombler"
+
+Bombler.kRadius = 0.25
+
+local kAnimationGraph = PrecacheAsset("models/alien/babbler/babbler.animation_graph")
+
+local kTargetMaxFollowRange = 30
+local kBabblerOffMapInterval = 1
+local kUpdateMoveInterval = 0.3
+local kMinJumpDistance = 5
+
+local oldOnInit = Bombler.OnInitialized
+function Bombler:OnInitialized()
+
+	oldOnInit(self)
+
+	if Server then
+		self:AddTimedCallback(Bombler.MoveRandom, kUpdateMoveInterval + math.random() / 5)
+	end
+
+end
+    
+
+function Bombler:TimedKill()
+
+	self:TriggerEffects("death", {effecthostcoords = Coords.GetTranslation(self:GetOrigin()) })
+	DestroyEntity(self)
+
+end
+
+
+function Bombler:GetIsClinged()
+    return false
+end							   
+
+
+
+
+local oldUpdateBabbler = Bombler.UpdateBabbler
+function Bombler:UpdateBabbler(deltaTime)
+
+	-- Kill babblers after a certain period
+	if Server then
+		if self.creationTime + kBomblerLifeTime  < Shared.GetTime() then
+			self:TimedKill()
+		end
+	end

```

Comment:
Why do you not use `Entity.AddTimedCallback`? Also, best practice is to always immediately call `return` in the main Update routine if you call a sub-method that destroys the entity. Otherwise, stuff may go haywire.